File: lib/models/invoice_detail.dart
class InvoiceDetail {
  final int invoiceId;
  final int productId;
  final String productName;
  final int quantity;
  final double unitPrice;
  final double discount;
  final double lineTotal;

  InvoiceDetail({
    required this.invoiceId,
    required this.productId,
    required this.productName,
    required this.quantity,
    required this.unitPrice,
    required this.discount,
    required this.lineTotal,
  });

  factory InvoiceDetail.fromJson(Map<String, dynamic> json) => InvoiceDetail(
        invoiceId: json['invoiceId'] ?? 0,
        productId: json['productId'] ?? 0,
        productName: json['productName'] ?? '',
        quantity: json['quantity'] ?? 0,
        unitPrice: (json['unitPrice'] ?? 0).toDouble(),
        discount: (json['discount'] ?? 0).toDouble(),
        lineTotal: (json['lineTotal'] ?? 0).toDouble(),
      );
}

File: Models/DataBase/InvoiceDetail.cs
namespace quan_ly_chuoi_cua_hang_tap_hoa_api.Models.DataBase
{
    public class InvoiceDetail
    {
        public int InvoiceId { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal Discount { get; set; }

        public virtual Invoice Invoice { get; set; }
        public virtual Product Product { get; set; }
    }
}
